.github/workflows/pre-commit-hooks.yaml

```diff
@@ -0,0 +1,20 @@
+name: pre-commit hooks
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  push:
+    branches:
+      - main
+      - develop
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
+
+      - uses: pre-commit/action@v3.0.1
```

Comment:
Place this before the pre-commit/action@v3.0.1 step.
Speeds up repeated CI runs significantly:
      - name: Cache pre-commit
        uses: actions/cache@v3
        with:
          path: ~/.cache/pre-commit
          key: ${{ runner.os }}-pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pre-commit-